### ISSUE TITLE
Support all branches for the `wait_for_workflows` command

### ISSUE THREAD

**Issue by roopakv** [roopakv/orbs#12]
Right now the `wait_for_workflows` command and job runs for only one branch. Make it support global (across all branches) and if possible multiple branches (using a regex)

**Comment by devgupta2607** [comment]
How can I make a contribution?

**Comment by roopakv** [comment]
Feel free tob send a PR my way. I am yet to write a contributing guide but it is pretty light weight right now

**Comment by jahn-mtg** [comment]
Any updates on this? Need this feature as well.

**Comment by roopakv** [comment]
@jahn-mtg let me take a look at this later today, might not be as hard. I will post back here when i have an update on how hard this is and the expected time

**Comment by roopakv** [comment]
@jahn-mtg this support is now there on v0.47.0 let me know if you have any questions, or find that it isnt working as you'd expect :)

**Comment by jahn-mtg** [comment]
@roopakv wow thank you so much for getting back to me on this! really appreciate it. I was hoping for the regex feature to be implemented, for perhaps something like "all branches except master"

**Comment by roopakv** [comment]
@jahn-mtg would love to understand the use case?

did you want to wait on all branches, but only block against workflows on non master branches? that seems like an edge case, and I'm not sure if i understand you correctly.

Could you explain your use case in detail in a new issue please? :)

**Comment by jahn-mtg** [comment]
@roopakv what we were looking for is the ability to block workflows if another branch has a workflow running EXCEPT when the workflow is occuring on master branch.

The use the case for this is that we have feature/branches all deploying to the same dev ENV and therefore don't want our automated tests to overlap, however master branch is pointed to an entirely different ENV (staging) so it should not block others

**Comment by roopakv** [comment]
@jahn-mtg you can achieve this by naming your workflows differently:

- feature branches:
workflow-name: "deploy-to-dev-env"
branch-to-consider: all

- master branch
workflow-name: "deploy-to-staging"
branch-to-consider: master

this way all feature branches will be queued one behind the other, master will be queued on itself alone.

Let me know if you think this wont solve your usecase

**Comment by jahn-mtg** [comment]
yea we can set it up that way. Thank you so much for the help and making the update! it was definitely helpful @roopakv
